### PR TITLE
New session

### PR DESCRIPTION
**Pull request by mrkn**
I rewrite `IRuby::Session` class to make writing tests easier than the current one.

## TODOs

- [x] Write ffi-rzmq session adapter
- [x] Write pyzmq session adapter

### CODE REVIEW

File: lib/iruby.rb [hunk 1]
```diff
@@ -15,18 +15,22 @@
 require 'iruby/utils'
 require 'iruby/display'
 require 'iruby/comm'
-require 'iruby/session/mixin'
 
-begin
-  require 'iruby/session/cztop'
-rescue LoadError
+if ENV.fetch('IRUBY_OLD_SESSION', false)
```
Comment (olleolleolle) [inline lib/iruby.rb]:
```suggestion
if ENV['IRUBY_OLD_SESSION']
```

Comment (olleolleolle) [reply lib/iruby.rb]: In this case, the default value offers no extra benefit - the absence of the String key returns `nil`, which is falsy.

Comment (mrkn) [reply lib/iruby.rb]: This is my intentional use of `Env.fetch`.

Comment (olleolleolle) [reply lib/iruby.rb]: It _does communicate_ intention. Carry on, don't mind my earlier input.

**Comment by olleolleolle** [comment]
Introducing Session and having adapters for it really makes this easier to talk and think about.

File: lib/iruby.rb (hunk 1 above)
Comment (mrkn) [reply lib/iruby.rb]: I'm sorry. My previous comment is not polite.

First, thank you very much for having an interest in IRuby development.  IRuby is not mature.  IRuby needs a lot of implementation improvements to catch up on Jupyter's evolutions.

I know ENV[...] is enough here as you described.  But does the replacement of ENV.fetch with ENV[...] improves the state of IRuby itself?  I don't think so.  I wanted to spend more time for improving the design and debugging of IRuby.  It's reason why i don't fix the line you pointed out in this pull-request.

I would appreciate if you understood that.

Comment (olleolleolle) [reply lib/iruby.rb]: Thank you, @mrkn, and I apologize for how I communicated.

**Pull request merged**